### ISSUE TITLE
Copy file name of files in read-only shared folder → deactivate drag & drop for read-only shares

### ISSUE THREAD

**Issue by kaysond** [nextcloud/server#10749]
If a directory is shared read-only with a link, there is no easy way to copy the file name. The text cannot be selected because dragging is enabled for the whole entry. It would be nice if the context (···) menu had an option for copying the filename.

**Comment by MorrisJobke** [comment]
> If a directory is shared read-only with a link, there is no easy way to copy the file name. The text cannot be selected because dragging is enabled for the whole entry. It would be nice if the context (···) menu had an option for copying the filename.

I don't like an separate entry for such a trivial thing. On macOS finder there is for example also no easy way to do this except choosing "rename" and then copy.

cc @nextcloud/designers

**Comment by kaysond** [comment]
> I don't like an separate entry for such a trivial thing. On macOS finder there is for example also no easy way to do this except choosing "rename" and then copy.

True. Windows is similar. But its not trivial because it's not possible to highlight the entry. When you're logged in, you can select rename and copy the file name. But if you're not logged in, the rename option doesn't exist. I think having an entry in that case (where the only other context menu option is "Download") would be fine.

**Comment by MorrisJobke** [comment]
> True. Windows is similar. But its not trivial because it's not possible to highlight the entry. When you're logged in, you can select rename and copy the file name. But if you're not logged in, the rename option doesn't exist. I think having an entry in that case (where the only other context menu option is "Download") would be fine.

Good point 👍

**Comment by jancborchardt** [comment]
> The text cannot be selected because dragging is enabled for the whole entry.

If the share is read-only, then drag & drop is useless anyway. It should be deactivated and then copying would be possible.

We should not add a specific entry for this in the menu. This is how it starts and then we end up with a list of dozens of useless entries. ;)

**Comment by kaysond** [comment]
That's a much better solution!

**Comment by szaimen** [comment]
I am going to close this since there doesn't seem to be a lot of interest and no respond since over 2 years. Please reopen if you still want to implement this.

**Comment by susnux** [comment]
Fixed with Nextcloud 31
